deploy.sh

```diff
@@ -3,34 +3,85 @@
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
 #
-# First check there is a valid license file for the Curity Identity Server
+# Check prerequisites
 #
 if [ "$LICENSE_FILE_PATH" == '' ]; then
-  echo '*** Please provide a LICENSE_FILE_PATH environment variable with the path to a Curity Identity Server license file'
+  echo '>>> Please provide a LICENSE_FILE_PATH environment variable with the path to a Curity Identity Server license file'
   exit 1
 fi
 
 export LICENSE_KEY=$(cat "$LICENSE_FILE_PATH" | jq -r .License)
 if [ "$LICENSE_KEY" == '' ]; then
-  echo '*** An invalid license file was provided for the Curity Identity Server'
+  echo '>>> An invalid license file was provided for the Curity Identity Server'
   exit 1
 fi
 
-##
-## Some MCP clients, like Claude, require HTTPS connections.
-## Therefore, create development SSL certificates for external URLs.
-##
-#./apigateway/certs/create.sh
-#if [ $? -ne 0 ]; then
-#  exit 1
-#fi
+base64url_decode() {
+  local len=$((${#1} % 4))
+  local result="$1"
+  if [ $len -eq 2 ]; then result="$1"'=='
+  elif [ $len -eq 3 ]; then result="$1"'=' 
+  fi
+  echo "$result" | tr '_-' '/+' | base64 --decode
+}
+
+LICENSE_PAYLOAD=$(base64url_decode $(echo $LICENSE_KEY | cut -d '.' -f 2))
+FEATURE=$(echo $LICENSE_PAYLOAD | jq -r '.Features[]  | select(.feature == "haapi")')
+if [ "$FEATURE" == '' ]; then
+  echo 'The license.json file does not include the HAAPI feature'
+  exit 1
+fi
+
+if [ -z "$NGROK_TOKEN" ]; then
+  echo '>>> Please set it to your ngrok authentication token in the NGROK_TOKEN environment variable'
+  exit 1
+fi
 
 #
-# Then run the deployment
+# Handle exposing the project with ngrok and support an explicit NGROK_DOMAIN parameter
+#
+NGROK_PID=$(pgrep ngrok)
+if [ -z "$NGROK_PID" ]; then
+
+  echo ">>> Start new instance of ngrok"
+
+  if [ -z "$NGROK_DOMAIN" ]; then
+    ngrok http 80 --authtoken $NGROK_TOKEN --config ngrok_no_ui.yml > /dev/null &
```

Comment:
Can't we add the `authtoken` to a config file as well? Ngrok seems to support multiple config files that it can merge, so there is no need to put the auth token in the `ngrok_no_ui.yml` and accidentally commit sensitive data. Instead, we can ignore the auth-token config file.
If we use env variables, we should use `NGROK_AUTHTOKEN` as that's the one used by ngrok. Some people may already have it in their bash.

Reply:
Let's do as a future refinement - I added a GitHub issue